app/controllers/TechnicalErrorController.scala

```diff
@@ -84,28 +83,37 @@ class TechnicalErrorController @Inject()(
           } yield BadRequest(view(formWithErrors, mode, retryAllowed)),
 
         value => {
-          personalDetailsValidationService.getPersonalDetailsValidationByNino(request.session.data.getOrElse("nino", StringUtils.EMPTY)).onComplete {
-            case Success(pdv) =>
-              auditService.audit(AuditUtils.buildAuditEvent(pdv.flatMap(_.personalDetails),
-                None,
-                "FindYourNinoOptionChosen",
-                pdv.map(_.validationStatus).getOrElse(""),
-                pdv.map(_.CRN.getOrElse("")).getOrElse(""),
-                Some(value.toString),
-                None,
-                None,
-                None,
-                None,
-                None
-              ))
-            case Failure(ex) => logger.warn(ex.getMessage)
-          }
           for {
             updatedAnswers <- Future.fromTry(request.userAnswers.set(TechnicalErrorPage, value))
             _ <- sessionRepository.set(updatedAnswers)
-            _ = if (value.toString == "tryAgain") tryAgainCountRepository.insertOrIncrement(request.userId)
-          } yield Redirect(navigator.nextPage(TechnicalErrorPage, mode, updatedAnswers))
+            pdvData <- personalDetailsValidationService.getPersonalDetailsValidationByNino(request.session.data.getOrElse("nino", StringUtils.EMPTY))
+          } yield {
+            val personalDetails = pdvData.flatMap(_.personalDetails)
+            val postcode: String = personalDetails.map(_.postCode.getOrElse(StringUtils.EMPTY)).getOrElse(StringUtils.EMPTY)
```

Comment:
This could be simplified a little 

`val postcode: String = personalDetails.flatMap(_.postCode).getOrElse(StringUtils.EMPTY)`